src/app/zones/views/ZoneListView.vue

```diff
@@ -13,103 +15,180 @@
         },
       ]"
     >
-      <div class="zones">
-        <MultizoneInfo v-if="store.getters['config/getMulticlusterStatus'] === false" />
+      <MultizoneInfo v-if="store.getters['config/getMulticlusterStatus'] === false" />
 
-        <div
-          v-else
-          class="stack"
+      <template v-else>
+        <DataSource
+          v-slot="{ data: zoneOverviewData, error, refresh }: ZoneOverviewCollectionSource"
+          :src="`/zones/zone-cps?size=${props.size}&page=${props.page}`"
         >
-          <div class="kcard-border">
-            <DataOverview
-              :selected-entity-name="entity?.name"
-              :page-size="PAGE_SIZE_DEFAULT"
-              :is-loading="isLoading"
-              :error="error"
-              :empty-state="EMPTY_STATE"
-              :table-data="tableData"
-              :table-data-is-empty="tableData.data.length === 0"
-              :show-warnings="tableData.data.some((item) => item.withWarnings)"
-              :next="nextUrl"
-              :page-offset="pageOffset"
-              :show-delete-action="env('KUMA_ZONE_CREATION_FLOW') === 'enabled'"
-              @delete-resource="toggleDeleteModal"
-              @table-action="loadEntity"
-              @load-data="loadData"
+          <DataSource
+            v-slot="{ data: allZoneIngressOverviewData }: AllZoneIngressOverviewCollectionSource"
+            :src="`/zones/all-zone-ingresses`"
+          >
+            <DataSource
+              v-slot="{ data: allZoneEgressOverviewData }: AllZoneEgressOverviewCollectionSource"
+              :src="`/zones/all-zone-egresses`"
             >
```

Comment:
**Comment**: This is not great. Ideally, we would be able to get multiple data sources out of one `DataSource` usage to mirror a `Promise.all([...])` use case (or is that somehow already possible?).

One problem here, besides it being annoying to write down like this in the template, is that we only use the `error` from the first block.

Reply:
Resolving this as it doesn’t apply to these components anymore as of https://github.com/kumahq/kuma-gui/pull/1209#discussion_r1268097990.